src/Quartz.Extensions.DependencyInjection/Simpl/MicrosoftDependencyInjectionJobFactory.cs

```diff
@@ -30,11 +30,16 @@ protected override IJob InstantiateJob(TriggerFiredBundle bundle, IScheduler sch
             //  Generate a scope for the job, this allows the job to be registered
             //	using .AddScoped<T>() which means we can use scoped dependencies 
             //	e.g. database contexts
-            var scope = serviceProvider.CreateScope();
+            var scope = ConfigureScope(bundle, scheduler);
             var (job, fromContainer) = CreateJob(bundle, scope.ServiceProvider);
             return new ScopedJob(scope, job, canDispose: !fromContainer);
         }
 
+        protected virtual IServiceScope ConfigureScope(TriggerFiredBundle bundle, IScheduler scheduler)
```

Comment:
I think it would be more understandable API if it's either `void ConfigureScope(IServiceScope scope)` which configures one already created or another option is  `IServiceScope CreateScope(IServiceProvider serviceProvider)` what do you think? CreateScope could be parameterless if services are not needed.

Reply:
The idea is to allow the configuration to be based on the job context. For example, I have a multi tenant application and I set the TenantId as a Job Data, so to configure the scope I need access to the context.

Reply:
Indeed the scope could be created and passed along with the context, that would avoid dispose problems for the caller.

Reply:
So would you like to refine the solution still?

Reply:
I've just pushed the changes